sphinx/themes/basic/layout.html

```diff
@@ -103,7 +103,25 @@
     <link rel="stylesheet" href="{{ pathto('_static/' + style, 1) }}" type="text/css" />
     <link rel="stylesheet" href="{{ pathto('_static/pygments.css', 1) }}" type="text/css" />
     {%- for cssfile in css_files %}
-    <link rel="stylesheet" href="{{ pathto(cssfile, 1) }}" type="text/css" />
+        {%- if cssfile.filename is defined %}
+            {%- set filename = cssfile.filename %}
+        {%- else %}
+            {%- set filename = cssfile %}
+        {%- endif %}
+        {%- if cssfile.alternate is defined %}
+            {%- set alt = cssfile.alternate %}
+        {%- endif %}
+        {%- if cssfile.title is defined and cssfile.title is string %}
+            {%- if cssfile.alternate is not defined %}
+                {%- set alt = True %}
+            {%- endif %}
+    <link rel="{% if alt %}alternate {% endif %}stylesheet" href="{{ pathto(filename, 1) }}" type="text/css" title="{{ cssfile.title }}" />
+        {%- else %}
+            {%- if cssfile.alternate is not defined %}
+                {%- set alt = False %}
+            {%- endif %}
+    <link rel="{% if alt %}alternate {% endif %}stylesheet" href="{{ pathto(filename, 1) }}" type="text/css" />
+        {%- endif %}
```

Comment:
IMO, css_files variable must not have multiple structure according to circumstances because it is difficult for sphinx users to understand and to tweak template. And also it is hard for maintainers to keep it stable.

BTW, I reconsidered about this change. Then I can't merge this for 1.4 release because side effect of this change is large for users who are using custom template that references to `css_files`. To resolve this concern, we should provide another variable.